cavern/src/main/java/org/opencadc/cavern/nodes/CephFSQuotaPlugin.java

```diff
@@ -0,0 +1,158 @@
+import java.nio.file.Path;
+
+
+/**
+ * QuotaPlugin implementation that can get and set specific values for interacting with CephFS.  This class relies
+ * on the Extended Attributes to be set on the appropriate folder.
+ */
+public class CephFSQuotaPlugin implements QuotaPlugin {
+    private static final Logger LOGGER = Logger.getLogger(CephFSQuotaPlugin.class);
+    private static final String NAMESPACE = "ceph";
+    private static final String QUOTA_ATTR_KEY = "quota.max_bytes";
+    private static final String RECURSIVE_SIZE_ATTR_KEY = "dir.rbytes";
+
+
+    @Override
+    public Long getBytesUsed(Path directory) {
+        LOGGER.debug("getBytesUsed: " + directory);
+        if (directory == null) {
+            throw new IllegalArgumentException("Cannot get size of null directory");
+        }
+
+        try {
+            final String recursiveBytesUsedValue =
+                    ExtendedFileAttributes.getFileAttribute(directory, CephFSQuotaPlugin.RECURSIVE_SIZE_ATTR_KEY,
+                                                            CephFSQuotaPlugin.NAMESPACE);
+            final long recursiveBytesUsed = Long.parseLong(recursiveBytesUsedValue);
+            LOGGER.debug("getBytesUsed: " + directory + ": OK");
+            return recursiveBytesUsed;
+        } catch (IOException ioException) {
+            LOGGER.warn("Unable to find recursive folder size of " + directory + " ("
+                        + CephFSQuotaPlugin.NAMESPACE + "." + CephFSQuotaPlugin.QUOTA_ATTR_KEY + ")");
+            return null;
+        }
+    }
+
+    /**
+     * Obtain the quota amount, in bytes, set at the highest level of the provided path.  This method will traverse,
+     * beginning at the first Path element, until it finds a directory containing the CephFS Quota attribute.
+     * Returning a null value means no quota is set, and users can write until the underlying storage is full.
+     *
+     * @param directory directory to check
+     * @return null if the directory input is null, or if no directory contains the quota attribute.
+     */
+    @Override
+    public Long getQuota(Path directory) {
+        LOGGER.debug("getQuota: " + directory);
+        if (directory == null) {
+            throw new IllegalArgumentException("Cannot check quota of null directory");
+        }
+
+        try {
+            final String quotaAttributeValue = ExtendedFileAttributes.getFileAttribute(directory,
+                                                                                       CephFSQuotaPlugin.QUOTA_ATTR_KEY,
+                                                                                       CephFSQuotaPlugin.NAMESPACE);
+            final Long quotaInBytes = StringUtil.hasText(quotaAttributeValue)
+                                      ? Long.parseLong(quotaAttributeValue) : null;
+            LOGGER.debug("getQuota: " + directory + ": OK");
+
+            return quotaInBytes;
+        } catch (IOException ioException) {
```

Comment:
NPE is covered but NumberFormatException is still possible (in principle)... and the message is not quite right now as it isn't "unable to find a folder" but a failed to read filesystem.

Here I am less sure about "warn" and return null. If deployer thinks they have configured quotas then a fail here should probably be a fail... so I think `throw new RuntimeException` is a correct internal fail.

Can probably use RuntimeException in setQuota as well.

iirc, ceph uses 0 as the value for "no quota"; I don't ecall if that is just a magic value for setting or if it might be returned...  that should never be returned from this method as-is. If there is no quota then the return value must be null.